app/js/playlist_NextBtn_anyPlaySelect.js

```diff
@@ -0,0 +1,173 @@
+function onClientLoad() {
+    gapi.client.load('youtube', 'v3', onYouTubeApiLoad);
+}
+
+// Called automatically when YouTube API interface is loaded
+function onYouTubeApiLoad() {
+    // This API key is intended for use only in this lesson.
+    gapi.client.setApiKey('AIzaSyCR5In4DZaTP6IEZQ0r1JceuvluJRzQNLE');
+}
+
+var inputKeySearch = "";
+
+function search() {
+    inputKeySearch = document.getElementById("keyWordSearch").value;
+    // Use the JavaScript client library to create a search.list() API call.
+    var request = gapi.client.youtube.search.list({
+        part: 'snippet',
+        q: inputKeySearch,
+        type: "video"
+    });
+
+    // Send the request to the API server,
+    // and invoke onSearchRepsonse() with the response.
+    request.execute(onSearchResponse);
+}
+
+// Called automatically with the response of the YouTube API request.
+function onSearchResponse(response) {
+  showResponse(response);
+}
+
+var catchSearchByKeywordId = [];
+var catchSearchByKeywordTitle = [];
+var catchSearchByKeywordDesc = [];
+
+// Helper function to display JavaScript value on HTML page.
+function showResponse(response) {
+  var responseString = JSON.stringify(response, '', 2);
+  document.getElementById('response').innerHTML += responseString;
+  for (var i =0; i<response.items.length; i++) {
+    catchSearchByKeywordTitle[i] = response.items[i].snippet.title;
+    catchSearchByKeywordDesc[i] = response.items[i].snippet.description;
+    catchSearchByKeywordId[i] = response.items[i].id.videoId;
+    document.getElementById("dispInfoTitle" + i).innerHTML = catchSearchByKeywordTitle[i];
+    document.getElementById("dispInfoDesc" + i).innerHTML = catchSearchByKeywordDesc[i];
+  }
+}
+
+function onYouTubeIframeAPIReady() {
+}
+
+var videoListIds = [];
+var videoCount = 0;
+
+function onPlayerStateChange(event) {
+  if (event.data == 0) {
+    if (videoCount >= videoListIds.length) {
+      videoCount = 0;
+      player.loadVideoById(videoListIds[videoCount]);
+      videoCount++;
+    }
+    else {
+      player.loadVideoById(videoListIds[videoCount]);
+      videoCount++;
+    }
+  }
+}
+
+function onPlayerReady() {
+}
+
+var player = {};
```

Comment:
@franciscotv do we need an empty object here?, or why you dont return the object from the function createPlayer?